### DISCUSSION TITLE
batch transfer images and yaml files to s3

### DISCUSSION THREAD

**Discussion by fredbouin** [statamic/cms#5831] [Q&A] [answered]
I have a bunch of image files I need to transfer from a local disk to s3, the problem is that they all have an associated yaml file with a lot of fields. I would much prefer not having to go back and manually input all the settings for each image. I tried uploading the images in Statamic cp and then logging in to aws and replacing the yaml files through the s3 interface but it doesn't work, if you click on the file after overwriting it seems to think it's an xml file.

So I'm wondering what the best method to batch transfer all images and associated yaml data to my bucket might be.

**Comment by ryanmitchell** [comment, marked as answer]
Why not use an FTP client that can connect to S3? Theres lots of them out there. You should be able to then just copy over everything to your S3 disk (yaml and all) and then update the disk in your filesystems config

**Reply by fredbouin** [reply]
ok this sounds like a possible solution, I'm not super savvy with these sort of things but i'll give it a whirl. Thanks for the tip.

**Reply by fredbouin** [reply]
ok this _almost_ works, this also happened when uploading images directly in the aws cp but basically everything works perfectly except for some reason the image preview in the assets panel within statamic's cp shows a broken link. Could it be a cache thing? clearing with php artisan config:cache does nothing.

Quick edit: what I'm noticing is when in the asset in statamic cp if I click "open in a new window" instead of loading the image in a new tab I get a new tab with this: 

```
<Error>
<Code>AccessDenied</Code>
<Message>Access Denied</Message>
<RequestId>VD17W2XS5TNXA0R7</RequestId>
<HostId>oU1Pe4R38Qpk8Da84xSc0b2iNqeJLobR3uIbmlZL3aI6qNr8R7UQgey9wOWuNBkW+zbvWglv2a8=</HostId>
</Error>
```

**Reply by ryanmitchell** [reply]
That’s due to your file system / s3 permissions - are you sure it’s public?

**Reply by fredbouin** [reply]
ahh got it, ok, the bucket is made public but I didn't realise I had to set every asset within it also as public. Anyway I think this pretty much solves my problem, thanks a lot.
